kategory-core/src/main/kotlin/kategory/typeclasses/Order.kt

```diff
@@ -0,0 +1,50 @@
+package kategory
+
+interface Order<F> : Eq<F>, Typeclass {
```

Comment:
What are the possible instances for this typeclass currently in the codebase, it would be nice to add them as part of this PR and include laws , test for instance discovery checking such as the ones existing in master for each of the datatypes that can support it.

Reply:
Currently I think only Number instances make sense. I've added them and the rest of your review remarks. Could you verify if I put them in the correct places?

Reply:
looks good, left a comment that needs to be addressed at https://github.com/kategory/kategory/pull/287/files#r140316364